identity-hub-did/src/main/java/org/eclipse/dataspaceconnector/identityhub/did/SignatureVerifier.java

```diff
@@ -0,0 +1,69 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.identityhub.did;
+
+import com.nimbusds.jwt.SignedJWT;
+import org.eclipse.dataspaceconnector.iam.did.crypto.credentials.VerifiableCredentialFactory;
+import org.eclipse.dataspaceconnector.iam.did.spi.resolution.DidPublicKeyResolver;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.result.Result;
+
+import java.text.ParseException;
+
+/**
+ * Class responsible for verifying JWT signatures.
+ */
+class SignatureVerifier {
+
+    private DidPublicKeyResolver didPublicKeyResolver;
+    private Monitor monitor;
+
+    private static final String VC_AUDIENCE = "identity-hub";
+
+    SignatureVerifier(DidPublicKeyResolver didPublicKeyResolver, Monitor monitor) {
+        this.didPublicKeyResolver = didPublicKeyResolver;
+        this.monitor = monitor;
+    }
+
+    /**
+     * Verifies if a JWT is really signed by the claimed issuer (iss field).
+     *
+     * @param jwt to be verified.
+     * @return if the JWT is signed by the claimed issuer.
+     */
+    boolean isSignedByIssuer(SignedJWT jwt) {
+        var issuer = getIssuer(jwt);
+        if (issuer.failed()) {
+            monitor.warning("");
+            return false;
+        }
+        var issuerPublicKey = didPublicKeyResolver.resolvePublicKey(issuer.getContent());
+        if (issuerPublicKey.failed()) {
+            return false;
+        }
+        var verificationResult = VerifiableCredentialFactory.verify(jwt, issuerPublicKey.getContent(), VC_AUDIENCE);
```

Comment:
I think the code of `verify` should be inlined to avoid using VerifiableCredentialFactory for this. See https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/issues/1714

In this way we can avoid use of `audience` claim and make any other necessary changes